src/Core/Framework/DataAbstractionLayer/Search/EntitySearchResult.php

```diff
@@ -9,8 +9,6 @@
 use Shopware\Core\Framework\Struct\StateAwareTrait;
 
 /**
- * @final
```

Comment:
Why do you need to extend this. Just saying Shopware does it is not a good explanation

Reply:
Updated my comment with: 

The EntitySearchResult class should be easily extendable. Third party plugin makers should be able to extend this class for their own entities. 
E.g.: A plugin has a ReceiptEntity that is also searchable and needs more functionality than the EntitySearchResult provides. In that case we would like to create a ReceiptSearchResult.

Reply:
Why you wan't them explictly in the SearchResult. Normally there are the collection classes for https://github.com/shopware/platform/blob/trunk/src/Core/Content/Product/ProductCollection.php#L18-L33